elasticsearch/src/main/java/org/hibernate/search/elasticsearch/util/impl/GsonHttpEntity.java

```diff
@@ -169,14 +169,27 @@ public Header getContentEncoding() {
 	public InputStream getContent() throws IOException {
 		//This could be implemented but would be sub-optimal compared to using produceContent().
 		//We therefore prefer throwing the exception so that we can easily spot unintended usage via tests.
-		throw new UnsupportedOperationException( "Not implemented! Expected to produce content only over produceContent()" );
+		throw new UnsupportedOperationException( "Not implemented! Expected to produce content only over produceContent(),"
+				+ " or writeTo(OutputStream) if blocking calls are acceptable for your use case." );
 	}
 
 	@Override
-	public void writeTo(OutputStream outstream) throws IOException {
-		//This could be implemented but would be sub-optimal compared to using produceContent().
-		//We therefore prefer throwing the exception so that we can easily spot unintended usage via tests.
-		throw new UnsupportedOperationException( "Not implemented! Expected to produce content only over produceContent()" );
+	public void writeTo(OutputStream out) throws IOException {
+		/*
```

Comment:
A nasty side-effect of implementing this method is that we won't notice if we actually start using this as primary serialization mechanism, so doing introducing a significant performance regression.
I'd prefer to throw an exception.